### PR TITLE
MM-60705 DB migration: added update_at column to draft table for mobile

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
#### Summary
Added `update_at` column to the draft table for mobile. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-60705

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.
- [ ] Have run E2E tests by adding label `E2E iOS tests for PR`.

#### Device Information
This PR was tested on: 
Andriod: 15 and iOS: 17.5

#### Release Note

```release-note
Added a new column `update_at` to the Drafts Table.
```

### CODE REVIEW

**Review by larkox — APPROVED** [review]

**Comment by larkox** [comment]
@Rajat-Dabade I think you are still missing the docs updates for this PR

**Comment by Rajat-Dabade** [comment]
@larkox will be updating it today.

**Review by yasserfaraazkhan — APPROVED** [review]

File: app/components/post_draft/index.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ const enhanced = withObservables(['channelId', 'rootId', 'channelIsArchived'], (
     }
 
     const draft = channelId.pipe(
-        switchMap((cId) => queryDraft(database, cId, rootId).observeWithColumns(['message', 'files', 'metadata']).pipe(
+        switchMap((cId) => queryDraft(database, cId, rootId).observeWithColumns(['update_at']).pipe(
```
Comment (enahum) [inline app/components/post_draft/index.ts:35]: Not part of this PR, correct?

File: app/database/operator/server_data_operator/transformers/post.ts [hunk 2]
```diff
@@ -112,6 +112,7 @@ export const transformDraftRecord = ({action, database, value}: TransformerArgs)
         draft.channelId = raw?.channel_id ?? '';
         draft.files = raw?.files ?? emptyFileInfo;
         draft.metadata = raw?.metadata ?? emptyPostMetadata;
+        draft.updateAt = raw.update_at;
```
Comment (enahum) [inline app/database/operator/server_data_operator/transformers/post.ts:115]:
```suggestion
        draft.updateAt = raw?.update_at || Date.now();
```

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/components/post_draft/index.ts (hunk 1 above)
Comment (Rajat-Dabade) [reply app/components/post_draft/index.ts:35]: Yes, reverting back the changes.

File: app/database/operator/server_data_operator/transformers/post.ts (hunk 2 above)
Comment (Rajat-Dabade) [reply app/database/operator/server_data_operator/transformers/post.ts:115]: Done.

File: app/database/operator/server_data_operator/transformers/post.ts [hunk 3]
```diff
@@ -112,6 +112,7 @@ export const transformDraftRecord = ({action, database, value}: TransformerArgs)
         draft.channelId = raw?.channel_id ?? '';
         draft.files = raw?.files ?? emptyFileInfo;
         draft.metadata = raw?.metadata ?? emptyPostMetadata;
+        draft.updateAt = raw.update_at || Date.now();
```
Comment (enahum) [inline app/database/operator/server_data_operator/transformers/post.ts:115]:
```suggestion
        draft.updateAt = raw.update_at ?? Date.now();
```

I guess 0 is a valid value

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/database/operator/server_data_operator/transformers/post.ts (hunk 3 above)
Comment (Rajat-Dabade) [reply app/database/operator/server_data_operator/transformers/post.ts:115]: Make sense.

**Review by enahum — APPROVED** [review]

**Pull request merged**